### ISSUE TITLE
Input mapping: shift modifier doesn't work with PAGEUP or PAGEDOWN

### ISSUE THREAD

**Issue by hsandt** [renpy/renpy#5156]
Platform: Linux Ubuntu 22.04

I have a conflict between a scrollable NVL box using scrollkeys pageup/pagedown and pageup/pagedown to rollback/rollforward. So I remapped rollback/rollforward to Shift+PageUp/PageDown. However, this didn't work at all (pressing said custom binding does nothing). I tried this:

```rpy
init python:
    config.keymap['rollback'].append('shift_any_K_PAGEUP')
    config.keymap['rollback'].append('shift_any_KP_PAGEUP')
    config.keymap['rollforward'].append('shift_any_K_PAGEDOWN')
    config.keymap['rollforward'].append('shift_any_KP_PAGEDOWN')
```

as well as variants such as `shift_K_PAGEUP` without `any`

`alt_` variants, on the contrary, work perfectly.

I suspect that Shift+PageUp/Down is interpreted as a different input (horizontal scrolling?), or maybe it's failing to be detected at a lower level.

**Comment by Gouvernathor** [comment]
For your particular use case I would suggest trying a `key "..." action Rollback()` in your screen, so that it remains a special case only for when the screen is displayed.

That aside, if the shift key binding works outside of that screen it means the screen is capturing it, otherwise it means the keysym doesn't work or isn't recognized.

If the viewport does capture shift+page keystrokes for horizontal scrolling, we should probably allow the pagekeys property to take "horizontal" and "vertical" as values, with similar behavior as with the scrollbars parameter.

**Comment by renpytom** [comment]
This works fine for me on Ubuntu 23.10. I'd check that you don't have something at the OS level that's eating the shift+PAGE events. You can use the xev tool to do that - focus it and hit the key, and see if it prints output. Either way, this doesn't seem to be a problem with Ren'Py.

**Comment by hsandt** [comment]
> For your particular use case I would suggest trying a key "..." action Rollback() in your screen, so that it remains a special case only for when the screen is displayed.

Thanks, I'll try per screen keymap. Ideally I'd like to enable the keymap specifically when scrollbar is visible but that may be  a bit complicated to implement even with https://lemmasoft.renai.us/forums/viewtopic.php?t=60347 (and possibly hard to user to understand the subtleties of input changing with context), so I may only use Alt+PageUp/Down in NVL mode where scrollbar is very rare, and normal PageUp/Down in History, About, etc. where scrolling is common.

I tried some keycapture but I see nothing special compared to other apps:

```sh
$ xwininfo
# click on Renpy game window

$ xev -id 0xb600014

# press Page Up

KeyPress event, serial 24, synthetic NO, window 0xb600014,
    root 0x553, subw 0x0, time 6796562, (550,414), root:(2194,1016),
    state 0x10, keycode 112 (keysym 0xff55, Prior), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

PropertyNotify event, serial 24, synthetic NO, window 0xb600014,
    atom 0x1b7 (_NET_WM_USER_TIME), time 6796563, state PropertyNewValue

# press Shift+PageUp

KeyPress event, serial 24, synthetic NO, window 0xb600014,
    root 0x553, subw 0x0, time 6830687, (399,396), root:(2043,998),
    state 0x11, keycode 112 (keysym 0xff55, Prior), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

PropertyNotify event, serial 24, synthetic NO, window 0xb600014,
    atom 0x1b7 (_NET_WM_USER_TIME), time 6830688, state PropertyNewValue
```

**Comment by renpytom** [comment]
Ah, there's your problem. 

You basically need to do: 

[code]
        config.keymap['viewport_pageup'] = [ 'K_PAGEUP', 'KP_PAGEUP'],
        config.keymap['viewport_pagedown'] = [ K_PAGEDOWN', 'KP_PAGEDOWN' ],
[/code]

Otherwise, pagekeys will eat the pageup and pagedown events, and the default keymap will never process them.

**Comment by hsandt** [comment]
I tried to add the non-any K_PAGEUP/DOWN to viewport_pageup/down, and even rollback/rollforward, but shift+pageup/down still doesn't work.

I'll wait for other people to try to repro on different platforms.

**Comment by renpytom** [comment]
Closing as unable to reproduce.

**Comment by hsandt** [comment]
Coming pretty late, but I rested on Renpy 8.5.3 on a brand new machine, and I couldn't repro, so confirmed it's all okay on my side.

**More details on my new tests below**

I had to slighty modify my code since it was surgically removing bindings 'anyrepeat_K_PAGEUP', 'anyrepeat_KP_PAGEUP' from config.keymap['viewport_pageup'] (resp. `_PAGEDOWN` for `viewport_pagedown`) instead of just replacing the keymap content, and due to some change in 7.8, but after that, I tried to bind config.keymap entries:

```
        viewport_pageup = [
            'shift_anyrepeat_K_PAGEUP',
            'shift_anyrepeat_KP_PAGEUP',
        ],
        viewport_pagedown = [
            'shift_anyrepeat_K_PAGEDOWN',
            'shift_anyrepeat_KP_PAGEDOWN',
        ],
```

and both worked, I could scroll up and down with shift+PageUp/Down!

No idea if this changed was fixed in some Renpy version or it was a machine-specific issue due to my full change of config, but I got my preferred mapping now. I also find the new `anyrepeat` clearer as I previously thought that `any` meant "any modifier", but that's actually `anymod`.
